### PR TITLE
Support for CLIP visual encoder

### PR DESCRIPTION
**Pull request by mukulkhanna**
## Motivation and Context

This PR enables support for a CLIP-trained ResNet visual encoder backbone for training and eval.

The code in this PR has been heavily sourced from the implementation here:

https://github.com/allenai/embodied-clip/tree/habitat

## Types of changes

- **\[Experiment\]** A pull request that add new features to the [habitat-baselines](/habitat-baselines/) training codebase. Experiments Pull Requests can be any size, must have smoke/integration tests and be isolated from the rest of the code. Your code additions should not rely on other habitat-baselines code. This is to avoid dependencies between different parts of habitat-baselines. You must also include a README that will document how to use your new feature or trainer. **You** will be the maintainer of this code, if the code becomes stale or is not supported often enough, we will eventually remove it.

## Checklist

- [x] My code follows the code style of this project.
- [ ] I have updated the documentation if required.
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes if required.

### CODE REVIEW

**Comment by mukulkhanna** [comment]
@Ram81 @naokiyokoyama Feel free to review.

File: habitat-baselines/habitat_baselines/rl/requirements.txt [hunk 1]
```diff
@@ -1,4 +1 @@
-moviepy>=1.0.1
```
[the remark this answers is not in the document]
Comment (mukulkhanna) [reply habitat-baselines/habitat_baselines/rl/requirements.txt]: That was a mistake, sorry. Readded. Thanks!

File: habitat-baselines/habitat_baselines/rl/requirements.txt [hunk 2]
```diff
@@ -1,4 +1 @@
...
-torch>=1.3.1
-protobuf==3.20.1
-tensorboard==2.8.0
+git+https://github.com/openai/CLIP.git@40f5484c1c74edd83cb9cf687c6ab92b28d8b656
```
[the remark this answers is not in the document]
Comment (mukulkhanna) [reply habitat-baselines/habitat_baselines/rl/requirements.txt:5]: Do you have any suggestions on how else we can do this?

By providing installation info/command through a friendly error message when someone tries to import this module without it being installed?

[the remark this answers is not in the document]
Comment (ASzot) [reply habitat-baselines/habitat_baselines/rl/requirements.txt:5]: Yes, I think an error message with instructions on how to download is best. See the `pygame` dependency in `interactive_play.py` for an example.

File: habitat-baselines/habitat_baselines/rl/ddppo/policy/resnet_policy.py [hunk 3]
```diff
@@ -409,22 +522,39 @@ def __init__(
...
-                nn.ReLU(True),
+            if not self.visual_encoder.is_blind:
+                self.visual_fc = nn.Sequential(
+                    nn.Flatten(),
+                    nn.Linear(
+                        np.prod(self.visual_encoder.output_shape), hidden_size
+                    ),
+                    nn.ReLU(True),
+                )
+        elif backbone.startswith("resnet50_clip"):
```
Comment (ASzot) [inline habitat-baselines/habitat_baselines/rl/ddppo/policy/resnet_policy.py:542]: Document which options are possible for `backbone`. You should also throw an error if it is an unrecognized option.

File: habitat-baselines/habitat_baselines/rl/requirements.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (mukulkhanna) [reply habitat-baselines/habitat_baselines/rl/requirements.txt]: Done!

File: habitat-baselines/habitat_baselines/rl/ddppo/policy/resnet_policy.py (hunk 3 above)
Comment (mukulkhanna) [reply habitat-baselines/habitat_baselines/rl/ddppo/policy/resnet_policy.py:551]: @ASzot Do you have any thoughts on this?

Comment (ASzot) [reply habitat-baselines/habitat_baselines/rl/ddppo/policy/resnet_policy.py:551]: Yes, you should explicitly check for any of them and document in the docstring. For example, if I pass `resnet50_clip_test` it should throw an error that this is not recognized.

**Review by ASzot — APPROVED** [review]

**Review by vincentpierre — APPROVED** [review]

**Review by vincentpierre — APPROVED** [review]

**Review by vincentpierre — APPROVED** [review]

**Pull request merged**